src/lib/account/evm.ts

```
@@ -9,13 +10,29 @@ import {
   supportsOptimism,
   supportsPolygon,
 } from '@shapeshiftoss/hdwallet-core'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import type { AccountMetadataById } from '@shapeshiftoss/types'
 import { assertGetEvmChainAdapter } from 'lib/utils/evm'
 
 import type { DeriveAccountIdsAndMetadata } from './account'
 
 export const deriveEvmAccountIdsAndMetadata: DeriveAccountIdsAndMetadata = async args => {
   const { accountNumber, chainIds, wallet } = args
+  // Initializes a cached EVM address, since all EVM ChainIds for a given accountNumber will yield the same address
+  // Unless we do this, Ledger will be spammed, which will make things slow at worst, and possibly break the connect button
+  const ledgerEvmAddress: string | null = await (async () => {
+    if (isLedger(wallet)) {
+      for (const chainId of chainIds) {
+        if (isEvmChainId(chainId)) {
+          const adapter = assertGetEvmChainAdapter(chainId)
+          return await adapter.getAddress({ accountNumber, wallet })
+        }
+      }
+    }
+    return null
+  })()
+    // if this throws for any reason, we don't want to make this whole async function reject
+    .catch(() => null)
```

i.e do not let this bubble up as unhandled promise rejection